### ISSUE TITLE
Jmx receiver got OutOfMemoryError 

### ISSUE THREAD

**Issue by junhuangli** [open-telemetry/opentelemetry-collector-contrib#21719]
### Component(s)

receiver/jmx

## Steps to Reproduce
Running for some time.(~8 hours if set collection_interval=1s, ~15 days if collection_interval=60s)

## Expected Result
Jvm/Cassandra metrics continue flowing

## Actual Result
Metrics stopped

### Collector version

v0.70.0

## Environment
NAME="CentOS Linux" VERSION="7 (Core)" ID="centos"

### OpenTelemetry Collector configuration

```yaml
---
receivers:
  jmx:
    jar_path: "/refinery/opentelemetry-jmx-metrics.jar"
    endpoint: localhost:7199
    target_system: cassandra,jvm
    collection_interval: 3s
    log_level: debug

  prometheus/internal:
    config:
      scrape_configs:
        - job_name: 'refinery-internal-metrics'
          scrape_interval: 10s
          static_configs:
            - targets: [ 'localhost:8888' ]
          metric_relabel_configs:
            - source_labels: [ __name__ ]
              regex: '.*grpc_io.*'
              action: drop

exporters:
  myexporter:
    retry_on_failure:
      enabled: true
      initial_interval: 5s
      max_interval: 30s
      max_elapsed_time: 300s
    myexporter:
      host: "myexporter.net"
      port: "9443"
      enable_mtls: true
      root_path: /etc/identity
      repo_dir_path: /etc/identity/client
      service_name: client
      gzip: true
 
processors:
  netmetadata:
    metrics:
      scopes: 
        service: refinery_tested
        subservice: "cassandra"
      
      tags:
        version: "1"
        k8s_pod_name: "test-cass-alrt-eap-c02-0"
        k8s_namespace: "dva-system"
        k8s_cluster: "collection-monitoring"
        device: "ip-10-11-11-11.us-west-2.compute.internal"
        substrate: "aws"
        account: "00000"
        region: "unknown"
        zone: "us-west-2b"
        falcon_instance: "dev1-uswest2"
        functional_domain: "monitoring"
        functional_domain_instance: "monitoring"
        environment: "dev1"
        environment_type: "dev"
        cell: "c02"
        service_name: "test-cass-alrt-eap"
        service_group: "test-shared"
        service_instance: "test-cass-alrt-eap-c02"
  memory_limiter/with-settings:
    check_interval: 1s
    limit_mib: 2000
    spike_limit_mib: 400
    limit_percentage: 0
    spike_limit_percentage: 0

  batch:
    timeout: 5s
    send_batch_size: 8192
    send_batch_max_size: 0

service:
  extensions: []
  telemetry:
    logs:
      development: false
      level: debug
    metrics:
      level: detailed
      address: localhost:8888
  pipelines:
    metrics:
      receivers: ["jmx"]
      processors: [memory_limiter/with-settings, batch, netmetadata]
      exporters: [myexporter]
    metrics/internal:
      receivers: ["prometheus/internal"]
      processors: [memory_limiter/with-settings, batch, netmetadata]
      exporters: [myexporter]
```

### Log output

```shell
2023-05-09T20:40:38.299Z	debug	subprocess/subprocess.go:287	java.lang.OutOfMemoryError: Java heap space	{"kind": "receiver", "name": "jmx", "pipeline": "metrics"}
2023-05-09T20:40:38.299Z	debug	subprocess/subprocess.go:287	SEVERE: Error gathering JMX metrics	{"kind": "receiver", "name": "jmx", "pipeline": "metrics"}
2023-05-09T20:40:38.298Z	debug	subprocess/subprocess.go:287	May 09, 2023 8:40:38 PM io.opentelemetry.contrib.jmxmetrics.JmxMetrics$1 run	{"kind": "receiver", "name": "jmx", "pipeline": "metrics"}
```

**Comment by junhuangli** [comment]
opentelemetry-jmx-metrics version 1.25.1

**Comment by rmfitzpatrick** [comment]
It's very likely this was resolved w/ instrument caching improvements that were provided in gatherer version 1.29.0 and made available in the v0.84.0 jmx receiver release. If you're still experiencing this, please open an issue in the gatherer's project since the receiver isn't in the jvm: https://github.com/open-telemetry/opentelemetry-java-contrib
